src/js/components/homepage/visualizations/treemap/TreeMapLabel.jsx

```diff
@@ -0,0 +1,74 @@
+/**
+ * TreeMapLabel.jsx
+ * Created by michaelbray on 6/2/17.
+ */
+
+import React from 'react';
+
+const propTypes = {
+    labelWidth: React.PropTypes.number,
+    labelPadding: React.PropTypes.number,
+    labelDistance: React.PropTypes.number,
+    currentX: React.PropTypes.number,
+    graphWidth: React.PropTypes.number
+};
+
+const defaultProps = {
+    labelWidth: 0,
+    labelPadding: 0,
+    labelDistance: 0,
+    currentY: 0,
+    currentX: 0,
+    graphHeight: 0,
+    graphWidth: 0,
+    size: 'small'
+};
+
+export default class TreeMapLabel extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            current: null
+        };
+
+        this.generatePoly = this.generatePoly.bind(this);
+    }
+
+    componentWillReceiveProps() {
+        this.generatePoly();
+    }
+
+    generatePoly() {
+        const leftLabelPos = this.props.labelWidth - this.props.labelPadding;
```

Comment:
`this.props` refers to the previous set of props, not the incoming set (since this is called by `componentWillReceiveProps`), thus there's a risk that the polyline is calculated using incorrect values.